cfme/tests/openstack/infrastructure/test_host_power_control.py

```diff
@@ -19,7 +19,8 @@ def host_collection(appliance):
 @pytest.fixture(scope='module')
 def host_on(host_collection, provider):
     try:
-        my_host_on = host_collection.all(provider)[0]
+        host_collection.filter({'provider': provider})
+        my_host_on = host_collection.all().pop()
```

Comment:
**REQUIRED**
Here and below
```python
my_host_on = provider.hosts.all().pop()
```

Reply:
Hi Dmitry,

Here is what i got when used my_host_on = provider.hosts.all().pop()


cfme/tests/openstack/infrastructure/test_host_power_control.py                                            [100%]

===================================================== ERRORS ======================================================
_____________________________ ERROR at setup of test_host_power_off[openstack_infra] ______________________________

host_collection = HostsCollection(filters={})
provider = OpenstackInfraProvider(endpoints={'default': <cfme.infrastructure.provider.ope...provider_data=None, api_version='Keystone v3', keystone_v3_domain_id='default')

    @pytest.fixture(scope='module')
    def host_on(host_collection, provider):
        try:
            host_collection.filter({'provider': provider})
>           my_host_on = provider.hosts.all().pop()

cfme/tests/openstack/infrastructure/test_host_power_control.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cfme/infrastructure/host.py:513: in all
    view = navigate_to(provider, 'Hosts') if provider else navigate_to(self, 'All')
../cfme_venv/lib/python2.7/site-packages/navmazing/__init__.py:113: in navigate
    return nav(cls_or_obj, self).go(0, *args, **kwargs)
cfme/utils/appliance/implementations/ui.py:553: in go
    self.check_for_badness(self.step, _tries, nav_args, *args, **kwargs)
cfme/utils/appliance/implementations/ui.py:400: in check_for_badness
    return fn(*args, **kwargs)
cfme/infrastructure/provider/__init__.py:387: in step
    self.prerequisite_view.entities.summary('Relationships').click_at('Hosts')
../cfme_venv/lib/python2.7/site-packages/widgetastic/widget.py:78: in wrapped
    return method(self, *new_args, **new_kwargs)
widgetastic_manageiq/__init__.py:618: in click_at
    return self._table.click_at(field_name)
../cfme_venv/lib/python2.7/site-packages/widgetastic/widget.py:78: in wrapped
    return method(self, *new_args, **new_kwargs)
widgetastic_manageiq/__init__.py:574: in click_at
    return self.get_field(field_name)[1].click()
../cfme_venv/lib/python2.7/site-packages/widgetastic/widget.py:78: in wrapped
    return method(self, *new_args, **new_kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = SummaryTable('.//table[./thead/tr/th[contains(@align, "left") and normalize-sp...n_widgets={}, assoc_column=None, rows_ignore_top=None, rows_ignore_bottom=None)
field_name = 'Hosts'

    def get_field(self, field_name):
        """Returns the table row or list of elements for rowspam case
                    of the field with this name.
            Args:
                field_name: Name of the field (left column)
    
            Returns:
                An instance of :py:class:`VanillaRow` or list of :py:class:`WebElement`
            """
        rowspan_path = './tbody//td[contains(text(), {})]/following-sibling::td'.format(
            quote(field_name))
        try:
            rowspan_attribute = self.browser.get_attribute('rowspan', self.row((0, field_name))[0])
        except IndexError:
>           raise NameError('Could not find field with name {!r}'.format(field_name))
E           NameError: Could not find field with name 'Hosts'

widgetastic_manageiq/__init__.py:520: NameError
______________________________ ERROR at setup of test_host_power_on[openstack_infra] ______________________________

host_collection = HostsCollection(filters={})
provider = OpenstackInfraProvider(endpoints={'default': <cfme.infrastructure.provider.ope...provider_data=None, api_version='Keystone v3', keystone_v3_domain_id='default')

    @pytest.fixture(scope='module')
    def host_off(host_collection, provider):
        try:
            host_collection.filter({'provider': provider})
>           my_host_off = provider.hosts.all().pop()

cfme/tests/openstack/infrastructure/test_host_power_control.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cfme/infrastructure/host.py:513: in all
    view = navigate_to(provider, 'Hosts') if provider else navigate_to(self, 'All')
../cfme_venv/lib/python2.7/site-packages/navmazing/__init__.py:113: in navigate
    return nav(cls_or_obj, self).go(0, *args, **kwargs)
cfme/utils/appliance/implementations/ui.py:553: in go
    self.check_for_badness(self.step, _tries, nav_args, *args, **kwargs)
cfme/utils/appliance/implementations/ui.py:400: in check_for_badness
    return fn(*args, **kwargs)
cfme/infrastructure/provider/__init__.py:387: in step
    self.prerequisite_view.entities.summary('Relationships').click_at('Hosts')
../cfme_venv/lib/python2.7/site-packages/widgetastic/widget.py:78: in wrapped
    return method(self, *new_args, **new_kwargs)
widgetastic_manageiq/__init__.py:618: in click_at
    return self._table.click_at(field_name)
../cfme_venv/lib/python2.7/site-packages/widgetastic/widget.py:78: in wrapped
    return method(self, *new_args, **new_kwargs)
widgetastic_manageiq/__init__.py:574: in click_at
    return self.get_field(field_name)[1].click()
../cfme_venv/lib/python2.7/site-packages/widgetastic/widget.py:78: in wrapped
    return method(self, *new_args, **new_kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = SummaryTable('.//table[./thead/tr/th[contains(@align, "left") and normalize-sp...n_widgets={}, assoc_column=None, rows_ignore_top=None, rows_ignore_bottom=None)
field_name = 'Hosts'

    def get_field(self, field_name):
        """Returns the table row or list of elements for rowspam case
                    of the field with this name.
            Args:
                field_name: Name of the field (left column)
    
            Returns:
                An instance of :py:class:`VanillaRow` or list of :py:class:`WebElement`
            """
        rowspan_path = './tbody//td[contains(text(), {})]/following-sibling::td'.format(
            quote(field_name))
        try:
            rowspan_attribute = self.browser.get_attribute('rowspan', self.row((0, field_name))[0])
        except IndexError:
>           raise NameError('Could not find field with name {!r}'.format(field_name))
E

Reply:
@jadhaj please rebase the PR and try again with `my_host_on = provider.hosts.all().pop()`